packages/apps/stock/src/screens/internalMoves/InternalMoveLineDetailsScreen.js

```diff
@@ -33,32 +33,54 @@ import {
   InternalMoveLineNotes,
   InternalMoveLineQuantityCard,
   InternalMoveLinePicker,
+  ProductTrackingNumberSelect,
 } from '../../components';
 import {fetchUnit} from '../../features/unitSlice';
 import {fetchProductWithId} from '../../features/productSlice';
-import {fetchInternalMoveLines} from '../../features/internalMoveLineSlice';
+import {
+  addTrackingNumber,
+  fetchInternalMoveLine,
+  fetchInternalMoveLines,
+} from '../../features/internalMoveLineSlice';
 import {fetchProductIndicators} from '../../features/productIndicatorsSlice';
 import StockMove from '../../types/stock-move';
 
+const trackingScanKey = 'tracking_internal-move-select';
+
 const InternalMoveLineDetailsScreen = ({navigation, route}) => {
   const {loadingProductFromId, productFromId} = useSelector(
     state => state.product,
   );
   const {activeCompany} = useSelector(state => state.user.user);
   const {productIndicators} = useSelector(state => state.productIndicators);
+  const {internalMoveLine, loadingInternalMoveLine} = useSelector(
+    state => state.internalMoveLine,
+  );
 
   const dispatch = useDispatch();
 
+  const trackingNumber = useMemo(
+    () => internalMoveLine?.trackingNumber ?? route.params.trackingNumber,
+    [internalMoveLine, route.params.trackingNumber],
+  );
+
+  const isTrackingNumberSelectVisible = useMemo(
+    () =>
+      productFromId?.trackingNumberConfiguration != null &&
+      trackingNumber == null,
+    [productFromId?.trackingNumberConfiguration, trackingNumber],
+  );
+
   useEffect(() => {
     dispatch(fetchUnit());
     if (
       route.params.internalMove != null &&
       route.params.internalMoveLine != null
     ) {
       dispatch(fetchInternalMoveLines(route.params.internalMove.id));
-      dispatch(fetchProductWithId(route.params.internalMoveLine.product.id));
+      dispatch(fetchProductWithId(route.params.internalMoveLine.product?.id));
```

Comment:
This will not worked anymore as you are asking for internalMoveLineId in props, please adapt code